stream-chat-android-client/src/main/java/io/getstream/chat/android/client/api/models/querysort/IQuerySort.kt

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) 2014-2022 Stream.io Inc. All rights reserved.
+ *
+ * Licensed under the Stream License;
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://github.com/GetStream/stream-chat-android/blob/main/LICENSE
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.getstream.chat.android.client.api.models.querysort
+
+public interface IQuerySort<T : Any> {
```

Comment:
Let's also add missing KDocs

Reply:
Let's also add missing KDocs

Reply:
We can have it named as "QuerySorter", "QuerySort", "IQuerySort" or something else. QuerySort would have the same name for the interface and the class, so we would have 2 different `QuerySort`. I personally like `QuerySorter`... `IQuerySort` is another option to avoid that the interface and class have the same name. Which do you prefer? Both are ok to me.

Reply:
I don't like `I` prefix as it's more like a C# convention. `QuerySorter` sounds much better

Reply:
Ideally, we should get rid of the existing `QuerySort`, cause it can be replaced by the new `QuerySortByReflection`.

We had 2 options:
1.  have both `class` & `interface` named `QuerySort` and resolve those by using the `as` keyword in kotlin. Customers are using a `class`, not an `interface`. So it's safe
2.  use a different name for `interface` until we delete the `class` version.

@leandroBorgesFerreira and I were discussing that, and just randomly picked up the name `IQuerySort`. 

P.S.
My personal preference is the option `1` to avoid massive changes in the future when we delete the `class` version and we will have to rename `IQuerySort` to `QuerySort`.

Reply:
@adasiewiczr we agreed to use `QuerySortByField` as a default implementation.
It is space & time friendly in terms of complexity.

Reply:
Let's use `QuerySorter` as decided in our daily.